dali/pipeline/operator/op_spec.h

```diff
@@ -135,20 +136,56 @@ class DLL_PUBLIC OpSpec {
    * @brief Add an instantiated argument with given name
    */
   DLL_PUBLIC inline OpSpec& AddInitializedArg(const string& name, Argument* arg) {
+    VerifyRenamedAfterDeprecated(name);
     DALI_ENFORCE(arguments_.find(name) == arguments_.end(),
         "AddArg failed. Argument with name \"" + name +
         "\" already exists. ");
-    arguments_[name].reset(arg);
-    return *this;
+    return SetInitializedArg(name, arg);
   }
 
   /**
    * @brief Sets or adds an argument with given name
+   *
+   * Routes the deprecated argument to appropriate name or drops them
    */
-  DLL_PUBLIC inline OpSpec& SetInitializedArg(const string& name, Argument* arg) {
-    arguments_[name].reset(arg);
+  DLL_PUBLIC inline OpSpec& SetInitializedArg(const string& arg_name, Argument* arg) {
+    if (schema_ && schema_->IsDeprecatedArg(arg_name)) {
+      const auto& deprecation_meta = schema_->DeprecatedArgMeta(arg_name);
+      // Argument is removed, and we can discard it
+      if (deprecation_meta.removed) {
+        return *this;
+      }
+      if (!deprecation_meta.renamed_to.empty()) {
+        const auto& new_arg_name = deprecation_meta.renamed_to;
+        DALI_ENFORCE(
+            arguments_.find(new_arg_name) == arguments_.end(),
+            make_string("Operator ", name(), " got an unexpected '", arg_name,
+                        "' deprecated argument when '", new_arg_name, "' was already provided."));
+
+        set_through_deprecated_arguments_[new_arg_name] = arg_name;
+        // Adjust the arg so it carries the proper name for serialization
+        if (arg->has_name()) {
+          arg->set_name(new_arg_name);
+        }
+        arguments_[new_arg_name].reset(arg);
+        return *this;
+      }
+    }
+    VerifyRenamedAfterDeprecated(arg_name);
+    arguments_[arg_name].reset(arg);
     return *this;
   }
+
+  /**
+   * @brief Check if the `arg_name` was already set through deprecated argument
+   */
+  DLL_PUBLIC inline void VerifyRenamedAfterDeprecated(const string& arg_name) {
```

Comment:
verify = check if something is true / authentic / real.

```suggestion
  DLL_PUBLIC inline void CheckIfSetWithAnotherName(const string& arg_name) {
```
or
```suggestion
  DLL_PUBLIC inline void EnforceUniqueArgName(const string& arg_name) {
```

Reply:
Done